### PR TITLE
Fix 'href' is missing in type error

### PR DESCRIPTION
**Pull request by duan602728596**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### 👻 What's the background?

1. Describe the source of requirement, like related issue link.

* typescript version: 3.3.3333
* ant-design: 3.15.2

If Button has type prop, it will report such an type error: 

```
Type '{ children: string; type: "primary"; }' is not assignable to type '(IntrinsicAttributes & IntrinsicClassA
ttributes<Button> & Pick<{ readonly href: string; readonly target?: string | undefined; readonly onClick?: ((event: MouseEvent<HTMLAnchorElement, MouseEvent>) => void) | undefined; ... 262 more ...; readonly onTransitionEndCapture?: ((
event: TransitionEvent<...>) => void) | unde...'.
  Property 'href' is missing in type '{ className: string; type: "danger"; shape: "circle"; icon: string; title: string; ghost: true; onClick: any; }' but required in type 'Pick<{ readonly href: string; readonly target?: string | undef
ined; readonly onClick?: ((event: MouseEvent<HTMLAnchorElement, MouseEvent>) => void) | undefined; readonly type?: "default" | ... 4 more ... | undefined; ... 261 more ...; readonly onTransitionEndCapture?: ((event: TransitionEvent<...
>) => void) | undefi...'.

button.d.ts(28, 5): 'href' is declared here.
```
Error sample code:

```typescript
import * as React from 'react';
import { Button } from 'antd';

class Demo extends React.Component {
  render(): React.ReactNode {

    return (
      <Button type="primary">button</Button>
    );
  }
}
```

2. Describe the problem and the scenario.

Fixed a type error when Button had type prop.

### 💡 Solution

1. How to fix the problem, and list final API implementation and usage sample if that is an new feature.

2. GIF or snapshot should be provided if includes UI/interactive modification.

### 📝 Changelog description

> Describe changes from userside, and list all potential break changes or other risks.

1. English description

Fixed a type error when Button has type prop.

2. Chinese description (optional)

修复了Button有type属性时的类型错误.

### ☑️ Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

**Comment by ztplz** [comment]
其实没必要  删了就行了 我们好像把它本来有的属性覆盖了

File: components/button/button.tsx [hunk 1]
```diff
@@ -63,7 +63,7 @@ export interface BaseButtonProps {
 }
 
 export type AnchorButtonProps = {
-  href: string;
+  href?: string;
```
Comment (ztplz) [inline components/button/button.tsx]: image
href本来nchor元素里就定义了 而且跟我们的没冲突

Comment (duan602728596) [reply components/button/button.tsx]: 其实主要是现在这个antd版本会提示这个错误

Comment (ztplz) [reply components/button/button.tsx]: 我的意思是我们现在的写法把原本有的href覆盖掉了 直接删掉href就行了

Comment (duan602728596) [reply components/button/button.tsx]: href和target都可以删掉了

Comment (ztplz) [reply components/button/button.tsx]: image
是的  onclick也可以删的  
image
这里y也有可以删的  
但是没必要  href删了就好了 接口以后慢慢整理吧   虽然一直就很乱  但是只是不合理而已 不会影响人

**Comment by afc163** [comment]
https://github.com/ant-design/ant-design/pull/15719
https://github.com/ant-design/ant-design/pull/15702

**Comment by afc163** [comment]
https://github.com/ant-design/ant-design/pull/15719
https://github.com/ant-design/ant-design/pull/15702

**Pull request closed without merging**